report_custom_filename/controllers/reports.py

```diff
@@ -39,7 +39,14 @@ def index(self, action, token):
         for report in report_xml.browse(report_ids):
             if not report.download_filename:
                 continue
-            objects = http.request.session.model(context['active_model'])\
+            # When you are on the form view of a customer and you
+            # click on the "Sales" button, context['active_model']
+            # is still res.partner (probably a bug...) but
+            # context['active_ids'] has the ID of the sale order
+            # cf https://github.com/OCA/reporting-engine/issues/56
+            # That's why we use report.model instead of context['active_model']
+            # together with context['active_ids']
+            objects = http.request.session.model(report.model)\
                 .browse(context['active_ids'])
```

Comment:
@hbrunn I can't, because I don't have any 'datas' key, not 'ids' key inside the action dict, nor any other information where I could get the ID of the object for the report except in context['active_ids']